src/Persistence/Array_.php

```diff
@@ -31,6 +29,52 @@ public function __construct(array $data = [])
      */
     protected $lastInsertIds = [];
 
+    public function __construct(array $data = [])
+    {
+        $this->seedData = $data;
+
+        // if there is no model table specified, then create fake one named 'data'
+        // and put all persistence data in there 1/2
+        if (count($this->seedData) > 0 && !isset($this->seedData['data'])) {
+            $rowSample = reset($this->seedData);
+            if (is_array($rowSample) && !is_array(reset($rowSample))) {
+                $this->seedData = ['data' => $this->seedData];
+            }
+        }
+    }
+
+    private function seedData(Model $model): void
+    {
+        if (isset($this->data[$model->table])) {
+            return;
+        }
+
+        $this->data[$model->table] = [];
+
+        if (isset($this->seedData[$model->table])) {
+            $rows = $this->seedData[$model->table];
+            unset($this->seedData[$model->table]);
+
+            foreach ($rows as $id => $row) {
+                $this->saveRow($model, $row, $id);
+            }
+        }
+
+        // for array persistence join which accept table directly (without model initialization)
+        foreach ($model->getFields() as $field) {
+            if ($field->hasJoin()) {
+                $join = $field->getJoin();
+                $joinTable = \Closure::bind(function () use ($join) {
```

Comment:
Why not simply `$joinTable = $join->foreign_table;` ?

Reply:
It is protected and I do not want to make it public because of this internal use.